src/core/config/parent_search.h

```diff
@@ -48,16 +48,29 @@
  *
  * Define as 1 to enable periodic parent search feature.
  *
- * When this feature is enabled an end-device/child (while staying attached) will periodically search for a possible
- * better parent and will switch parent if a better one is found.
+ * When this feature is enabled, an end device/child (while staying attached) periodically searches for a possible
+ * better parent and switches parents if a better one is found.
  *
- * The child will periodically check the average RSS value for the current parent, and only if it is below a specific
- * threshold, a parent search is performed. The `OPENTHREAD_CONFIG_PARENT_SEARCH_CHECK_INTERVAL` specifies the
- * check interval (in seconds) and `OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD` gives the RSS threshold.
+ * Every `OPENTHREAD_CONFIG_PARENT_SEARCH_CHECK_INTERVAL` seconds, the child checks its average RSS to its current
+ * parent. Only if the average RSS is below a configured threshold (`OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD`)
+ * does the device start a parent search process.
```

Comment:
Why not always try to switch parents if there is a neighboring router that has a much better signal strength?

Reply:
This rule applies to both MTD and FTD.

For MTD (SED), we used this to skip the entire search process and save battery life if the child is happy with its current parent (has a decent RSSI). For FTD, I agree we can consider relaxing this. 

The way I would suggest is to use different (default) threshold values. By setting `OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD` to a large value on FTD, we can effectively disable this check. This approach offers flexibility while reusing the same logic and code.

Thoughts?

Reply:
Yes, having a different default values makes sense. I would suggest setting the FTD to disable this check.

Reply:
Updated the PR. Opted to fully remove the current "Parent RSS" check for FTD child.